src/dos/drive_fat.cpp

```diff
@@ -1007,12 +1021,20 @@ bool fatDrive::FileCreate(DOS_File **file, char *name, uint16_t attributes) {
 
 	/* Check if file already exists */
 	if(getFileDirEntry(name, &fileEntry, &dirClust, &subEntry)) {
+		if (fileEntry.attrib & DOS_ATTR_READ_ONLY) {
+			DOS_SetError(DOSERR_ACCESS_DENIED);
+			return false;
+		}
+
 		/* Truncate file */
 		if (fileEntry.loFirstClust != 0) {
 			deleteClustChain(fileEntry.loFirstClust, 0);
 			fileEntry.loFirstClust = 0;
 		}
 		fileEntry.entrysize = 0;
+		fileEntry.attrib    = (uint8_t)(attributes & 0xff);
```

Comment:
My first comment is to replace the C cast with a C++ `static_cast`. 

But curious: do we expect `attributes` to have some flags higher than the first byte, and so we just sheer them off here with `& 0ff`? 

But if the answer is no - that `attributes` really should only have values limited within the 8-bit range, then I would suggest using a `check_cast<uint8_t>(attributes)` instead - and avoid the sheering.